src/Provider.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from "react"
+import { shallowEqual } from "./utils/utils"
+import { IValueMap } from "./inject"
+
+export const MobXProviderContext: React.Context<any> = React.createContext({})
+
+export interface ProviderProps extends IValueMap {
+    children: React.ReactNode
+}
+
+export function Provider(props: ProviderProps): React.ReactElement {
```

Comment:
`React.FC` has one drawback, which is that it does not require the explicit specification of children :( For this reason, we plan to prohibit using React.FC on our project.

Reply:
We do not need to specify the return value of this function, so we can use:
```
export function Provider(props: ProviderProps) {
```
This version is more readable than `React.FC`. So in my opinion, we should leave the function declaration as is.

Reply:
I'm cool with that, just removing the return type.

Reply:
Well, in this case, it's fine I suppose, but the advantage of `React.FC` is that it guards a correct return type too. It's now allowed to return `undefined | void` and without explicit return type, it won't be guarded.

That's why we (on the job) use `React.FC` and we have own set of type aliases to cover for children. We can do eg. `type TProps = NoChildren & { somethingElse: string }` and it will complain if some children are passed. And vice versa with `SingleChild`, `SomeChildren` and combination with built-in `Required` works really well.